app/code/community/Bolt/Boltpay/Model/Order.php

```diff
@@ -329,7 +279,69 @@ public function createOrder($reference, $sessionQuoteId = null, $isPreAuthCreati
+     * @param $transaction
+     * @param $isPreAuthCreation
+     * @return bool
+     * @throws Exception
+     */
+    protected function applyShipmentToQuoteFromTransaction($immutableQuote, $transaction, $isPreAuthCreation){
+        $packagesToShip = @$transaction->order->cart->shipments;
+
+        if (!$packagesToShip) {
+            return false;
+        }
+
+        benchmark( "Applying shipping" );
+        $shippingAddress = $immutableQuote->getShippingAddress();
+        $shippingMethodCode = null;
+
+        /** @var Bolt_Boltpay_Model_ShippingAndTax $shippingAndTaxModel */
+        $shippingAndTaxModel = Mage::getModel("boltpay/shippingAndTax");
+        $shouldRecalculateShipping = ((bool) $this->boltHelper()->getExtraConfig("recalculateShipping")) || !$isPreAuthCreation; # false by default
+
+        benchmark( "Applying shipping - Applying shipping address data" );
+        $shippingAndTaxModel->applyBoltAddressData($immutableQuote, $packagesToShip[0]->shipping_address, $shouldRecalculateShipping);
+        benchmark( "Finished applying shipping - Applying shipping address data" );
+
+        $shippingMethodCode = $packagesToShip[0]->reference;
+
+        if (!$shippingMethodCode) {
+            benchmark( "Applying shipping - Collecting shipping rates, legacy" );
+            // Legacy transaction does not have shipments reference - fallback to $service field
+            $shippingMethod = $packagesToShip[0]->service;
+
+            $this->boltHelper()->collectTotals($immutableQuote, $shouldRecalculateShipping);
+
+            $rates = $shippingAddress->getAllShippingRates();
 
+            foreach ($rates as $rate) {
+                if ($rate->getCarrierTitle() . ' - ' . $rate->getMethodTitle() === $shippingMethod
+                    || (!$rate->getMethodTitle() && $rate->getCarrierTitle() === $shippingMethod)) {
+                    $shippingMethodCode = $rate->getCarrier() . '_' . $rate->getMethod();
+                    break;
+                }
+            }
+            benchmark( "Finished applying shipping - Collecting shipping rates, legacy" );
+        }
+
+        if ($shippingMethodCode) {
+            $shippingAndTaxModel->applyShippingRate($immutableQuote, $shippingMethodCode, $shouldRecalculateShipping);
+        } else {
+            $errorMessage = $this->boltHelper()->__('Shipping method not found');
+            $metaData = array(
+                'transaction'   => $transaction,
+                'rates' => $this->getRatesDebuggingData($rates),
+                'service' => $shippingMethod,
+                'shipping_address' => var_export($shippingAddress->debug(), true),
+                'quote' => var_export($immutableQuote->debug(), true)
+            );
+            $this->boltHelper()->logWarning($errorMessage);
+            $this->boltHelper()->notifyException(new Exception($errorMessage), $metaData);
+        }
+        benchmark( "Finished applying shipping" );
+    }
```

Comment:
If you have declared that you are going to return a boolean, then here, we are expecting a boolean to be returned.  Perhaps you are wanting to return true stating that a shipping method has been applied.  However, I advise instead to remove the declared return type and instead of returning false on line 293 in the early return, just simply return.  This way, you do not have to add any new logic.